File: Payment.java
public class Payment {
    private int paymentID;
    private String paymentMethod;
    private int userID;
    private float amount; 
    private String state;
    private Ticket ticketID;

    public boolean processPayment(String paymentMethod,float price,int userID,String accountNumber,String password){
        return false;
        
    }
    public String refundPayment(String state){
        return state;
        
    }
    public String getPaymentStatus(){
        return state;
        
    }
}

File: Payment.php
<?php
include_once 'database.php';
abstract class Payment{
    protected $coon;
    protected $paymentId;
    protected $userId;
    protected $amount;
    protected $state;
    protected $ticketId;
    protected $method;

    public function __construct($db){
        $this->conn = $db;
    }
    ///دالة معالجة الدفع يتم تنفيذها في كلايات الابناء 
    abstract public function processPayment();

    ////دالة ارجاع المبلغ في حالة الغاء الحجز
    public function refoundPayment($state){
        //TODO
    }
}
